### PR TITLE
update-template

### PR DESCRIPTION
**Pull request by georgewoodhead**
## Description
- Update to using the latest workflows and templates from the Unified package
- Add base models and variables to get started with, we should be encouraging all our dbt packages to make use of the Utils base macros incremental framework
- Add integration test directory, again all dbt packages should have this

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Checklist
- [ ] 🎉 I have verified that these changes work locally
- [ ] 💣 Is your change a breaking change?
- [ ] 📖 I have updated the CHANGELOG.md

### Added tests?
- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?
- [x] 📓 internal package docs (ymls, macros, readme, if applicable)
- [ ] 📕 I have raised a [Snowplow documentation](https://github.com/snowplow/documentation) PR if applicable (Link here if required)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: dbt_project.yml
Comment (rlh1994) [inline dbt_project.yml:1]: Think you're missing the on-run-end hook to update the manifest table?

File: integration_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Good-to-knows:
+
+There are certain exceptions to how different warehouses process data and in places we had to adjust the integration test to work around that. Here's a list of things to keep in mind:
+
+- the non-deterministic nature of row_number() function for Redshift/Postgres/Databricks means that we had to hard-code actuals and expected models for cases where we are testing duplicate rows with exact same results / window
+- postgres / redshift needing the array format of : (within sessions_expected)
+- bigquery handling of snowplow_utils.timestamp_diff() - absolute_time_in_s changes
+- rotating domain_userid per session is hard-coded in the integration test expectations, when run in one batch the user_identifier differs: 2e340eb6e94820ea8369c0174c612260d1cfe9d41f0fe46268994e28d9c0bbf17
+0e9ab97b5d9d9a174112df13fe9c44788af3ac9088a8b41e0998d92a8b4b5a4fc
+- same with the number of quarantined sessions
```
Comment (rlh1994) [inline integration_tests/README.md]: Maybe rephrase these as things to check for and suggested fixes, because these were specific cases

File: models/base/manifest/<YOUR PACKAGE NAME>_base_sessions_lifecycle_manifest.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    materialized='incremental',
+    unique_key='session_identifier',
+    upsert_date_key='start_tstamp',
+    sort='start_tstamp',
+    dist='session_identifier',
+    partition_by = snowplow_utils.get_value_by_target_type(bigquery_val={
+      "field": "start_tstamp",
+      "data_type": "timestamp"
+    }, databricks_val='start_tstamp_date'),
+    cluster_by=snowplow_utils.get_value_by_target_type(bigquery_val=["session_identifier"], snowflake_val=["to_date(start_tstamp)"]),
```
Comment (rlh1994) [inline models/base/manifest/<YOUR PACKAGE NAME>_base_sessions_lifecycle_manifest.sql]: Can we swap this out for a macro instead please, as a customer wanted that in unified

File: macros/allow_refresh.sql
Comment (rlh1994) [inline macros/allow_refresh.sql:1]: See this discussion https://snowplow.slack.com/archives/C047LRF74SV/p1707930996528249

File: models/base/manifest/<YOUR PACKAGE NAME>_base_sessions_lifecycle_manifest.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    days_late_allowed=var('snowplow__days_late_allowed', 3),
+    max_session_days=var('snowplow__max_session_days', 3),
+    app_ids=var('snowplow__app_id', []),
+    snowplow_events_database=var('snowplow__database', target.database) if target.type not in ['databricks', 'spark'] else var('snowplow__databricks_catalog', 'hive_metastore') if target.type in ['databricks'] else var('snowplow__atomic_schema', 'atomic'),
+    snowplow_events_schema=var('snowplow__atomic_schema', 'atomic'),
+    snowplow_events_table=var('snowplow__events_table', 'events'),
+    event_limits_table='<YOUR PACKAGE NAME>_base_new_event_limits',
+    incremental_manifest_table='<YOUR PACKAGE NAME>_incremental_manifest'
+) %}
+
```
Comment (rlh1994) [inline models/base/manifest/<YOUR PACKAGE NAME>_base_sessions_lifecycle_manifest.sql]: This is missing the package name argument, but we're also missing it in unified... I _think_ this only means we maybe are updating the sessions manifest table even when there are no new events, but i'm not 100% sure...

**Review by rlh1994 — CHANGES_REQUESTED** [review]
I didn't read the docs stuff but it's pretty much there, nice work!

File: dbt_project.yml
Comment (rlh1994) [reply dbt_project.yml:1]: Also the on-run-start hook to delete models

File: models/base/manifest/<YOUR PACKAGE NAME>_base_sessions_lifecycle_manifest.sql (hunk 3 above)
Comment (georgewoodhead) [reply models/base/manifest/<YOUR PACKAGE NAME>_base_sessions_lifecycle_manifest.sql]: Added it in ✔️ . Nice spot on it missing in unified too! Yeah as far as I can follow the `is_run_with_new_events` macro will currently always return true because the package name doesn't match up

**Review by rlh1994 — APPROVED** [review]
